dotnet/src/SemanticKernel.Abstractions/SkillDefinition/SKFunctionContextParameterAttribute.cs

```diff
@@ -6,22 +6,36 @@
 namespace Microsoft.SemanticKernel.SkillDefinition;
 
 /// <summary>
-/// Attribute to describe the parameters required by a native function.
-///
-/// Note: the class has no ctor, to force the use of setters and keep the attribute use readable
-/// e.g.
-/// Readable:     [SKFunctionContextParameter(Name = "...", Description = "...", DefaultValue = "...")]
-/// Not readable: [SKFunctionContextParameter("...", "...", "...")]
+/// Attribute to describe a parameters used by a native function.
 /// </summary>
-[AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+/// <remarks>
+/// The attribute may be applied to parameters to a method attributed with <see cref="SKFunctionAttribute"/>
+/// to provide additional information for that parameter, including a description, an optional default value,
+/// and an optional name that may be used to override the name of the parameter as specified in source code.
+/// The attribute may also be applied to a method itself to describe a context variable that is not specified
+/// in the method's signature, in which case a name is required to identify which context variable is being described.
+/// </remarks>
+[AttributeUsage(AttributeTargets.Method | AttributeTargets.Parameter, AllowMultiple = true)]
 public sealed class SKFunctionContextParameterAttribute : Attribute
```

Comment:
This is a quite verbose attribute, especially to be specified multiple times. We might want to consider renaming:
- SKFunctionParameter
- SKParameter
- SKParam